classworks/ivan-drobysh/15_websockets/demo/src/client/src/components/chat/chat.jsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import { sendMessage, pushNewMessage } from 'resources/message/message.actions';
+import { getUsername } from 'resources/user/user.selectors';
+import { setRoomId } from 'resources/room/room.actions';
+import { getRoomId } from 'resources/room/room.selectors';
+import { loadMessages, deleteMessage, deleteMessageSelf } from 'resources/message/message.actions';
+import { getMessages } from 'resources/message/message.selectors';
+import { connect } from 'react-redux';
+import Message from './components/message/index';
+import io from 'socket.io-client';
+
+const socket = io();
+
+let timerId;
+import './chat.styles';
+
+class Chat extends React.Component {
+  onClick = (messageId) => () => {
+    return this.props.deleteMessage({ messageId });
+  }
+  componentWillMount() {
+    this.state = {
+      content: '',
+      roomId: this.props.roomId 
+    };
+
+    this.props.loadMessages({ roomId: this.state.roomId });
+    socket.emit('subscribe', { roomId: this.state.roomId || 'public'  });
+
+    socket.on('message:sent', message => {
+      this.props.pushNewMessage(message);
+    });
+
+    socket.on('message:remove', messageId => {
+      this.props.deleteMessageSelf({ messageId });
+    });
+
+    socket.on('typing', userId => {
+      if(userId)
```

Comment:
it's unclear what is the case what is covered by this if. consider creating a variable which name will tell what does it mean